src/modules/jcms_article/src/FetchArticleVersions.php

```diff
@@ -68,24 +70,36 @@ public function getArticleVersions(string $id): ArticleVersions {
    * @return \Psr\Http\Message\ResponseInterface
    * @throws \TypeError
    */
-  function requestArticleVersions(string $id): ResponseInterface {
+  private function requestArticleVersions(string $id): ResponseInterface {
     $options = [
       'headers' => [
         'Authorization' => Settings::get('jcms_article_auth_unpublished'),
       ],
-      'http_errors' => FALSE,
     ];
     $url = $this->formatUrl($id, $this->endpoint);
-    $response = $this->client->get($url, $options);
-    if ($response instanceof ResponseInterface) {
-      \Drupal::logger('jcms_article')
-        ->notice(
-          'Article versions have been requested @url with the response: @response',
-          ['@url' => $url, '@response' => \GuzzleHttp\Psr7\str($response)]
-      );
-      return $response;
+    try {
+      $response = $this->client->get($url, $options);
+      if ($response instanceof ResponseInterface) {
+        \Drupal::logger('jcms_article')
+          ->notice(
+            'Article versions have been requested @url with the response: @response',
+            ['@url' => $url, '@response' => \GuzzleHttp\Psr7\str($response)]
+          );
+        return $response;
+      }
+      throw new \TypeError('Network connection interrupted on request.');
```

Comment:
It's not a `TypeError`.

Reply:
I'm thinking we can just remove this. If `ResponsInterface` is not returned then an exception will be thrown by guzzle and it can certainly handle lose of connection.